tests/rptest/services/redpanda.py

```diff
@@ -439,17 +445,15 @@ def to_cli(self, *, dedicated_node):
         else:
             memory_mb = self._memory_mb
 
-        if self._bypass_fsync is None and not dedicated_node:
-            bypass_fsync = True
-        else:
-            bypass_fsync = self._bypass_fsync
+        bypass_fsync = True if dedicated_node else self._bypass_fsync
+
+        args: list[str] = []
+
+        if not dedicated_node or self._force_overprisioned:
+            args.append("--overprovisioned")
 
-        args = []
         if not dedicated_node:
-            args.extend([
-                "--kernel-page-cache=true", "--overprovisioned ",
-                "--reserve-memory=0M"
-            ])
+            args.extend(["--kernel-page-cache=true", "--reserve-memory=0M"])
```

Comment:
didn't know we used this

Reply:
Yeah I guess the idea is to speed things up on slow disks in docker, though I'm not sure how much it speeds things up in practice once you've disable fsync already. I believe it means that short tests don't even write anything to disk at all: RP comes up, writes stuff which is cached in the page cache, then RP is torn down and the data directory deleted so potentially the pages are never flushed at all.